### PR TITLE
Add doc for custom fragments

### PR DESCRIPTION
**Pull request by cstoidner**
## Proposed changes

Reference documentation for custom fragments

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: docs/src/references/custom-fragments.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+## Custom-Fragments on MQTT
+
+All **custom-fragments** are provided and consumed via MQTT.
+
+
+#### Topics
+
+```cjson
+/tedge/inventory/<device-id>/<identifider>
```
Comment (PradeepKiruvale) [inline docs/src/references/custom-fragments.md:18]:
```suggestion
tedge/inventory/<device-id>/<identifider>
```

File: docs/src/references/custom-fragments.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+```
+
+|Reference     | Description|
+| ------------ | ---------- |
+|`device-id`   | if the custom-fragment is meant for a **child-device**, it is the corresponding **child-id**; or `main` if it is meant for the **main-device**|
```
Comment (didier-wenzek) [inline docs/src/references/custom-fragments.md:23]: Using an explicit `main` id is a really good idea, removing many issues we currently have with topics that need to be parsed differently for the main and for child devices.

**Review by didier-wenzek — COMMENTED** [review]
One needs to consider how this approach fits or replaces the current file-based implementation for fragment. Looking at the issue https://github.com/thin-edge/thin-edge.io/issues/1296, one  also needs to see how some fragments can be dynamically updated by thin-edge itself using a computation rule provided by the user (e.g device firmware version or up-time).

**Comment by didier-wenzek** [comment]
Closing as this is under full redesign.

**Pull request closed without merging**